src/uwtools/tests/drivers/test_chgres_cube.py

```diff
@@ -101,6 +107,14 @@ def ready(x):
     assert isinstance(f90nml.read(dst), f90nml.Namelist)
 
 
+def test_ChgresCube_namelist_file_fails_validation(caplog, driverobj):
+    log.setLevel(logging.INFO)
+    driverobj._driver_config["namelist"]["update_values"]["config"]["convert_atm"] = "string"
+    with patch.object(chgres_cube, "file", new=ready):
+        driverobj.namelist_file()
+    assert logged(caplog, "  'string' is not of type 'boolean'")
```

Comment:
Here and below, there are new tests that exercise the namelist-fails-final-validation case.